charmcraft/providers/_buildd.py

```diff
@@ -40,6 +47,45 @@ class CharmcraftBuilddBaseConfiguration(bases.BuilddBase):
 
     compatibility_tag: str = f"charmcraft-{bases.BuilddBase.compatibility_tag}.0"
 
+    def _setup_charmcraft(self, *, executor: Executor) -> None:
+        """Install Charmcraft in target environment.
+
+        On Linux, the default behavior is to inject the host snap into the target
+        environment.
+
+        On other platforms, the Charmcraft snap is installed from the Snap Store.
+
+        When installing the snap from the Store, we check if the user specifies a
+        channel, using CHARMCRAFT_INSTALL_SNAP_CHANNEL=<channel>.  If unspecified,
+        we use the default determined by get_managed_environment_snap_channel_default().
+        On Linux, the user may specify this environment variable to force Charmcraft
+        to install the snap from the Store rather than inject the host snap.
+
+        :raises BaseConfigurationError: on error.
+        """
+        snap_channel = get_managed_environment_snap_channel()
+        if snap_channel is None and sys.platform != "linux":
+            snap_channel = get_managed_environment_snap_channel_default()
+
+        if snap_channel:
+            try:
+                snap_installer.install_from_store(
+                    executor=executor, snap_name="charmcraft", channel=snap_channel, classic=True
+                )
+            except snap_installer.SnapInstallationError as error:
+                raise bases.BaseConfigurationError(
+                    brief="Failed to install Charmcraft snap from store into target environment.",
```

Comment:
Please include the channel in this error, it will be helpful to debug when the channel is mispelled in the env var.

Reply:
Done!

Reply:
thanks